src/app/@ansyn/ansyn/modules/status-bar/components/search-panel/search-panel.component.ts

```diff
@@ -96,25 +95,32 @@ export class SearchPanelComponent implements OnInit, OnDestroy {
 	ngOnInit() {
 	}
 
-	toggleDataInputFilter() {
-		this.dataInputFilterExpand = !this.dataInputFilterExpand;
-	}
-
-	toggleTimePicker() {
-		this.timePickerExpand = !this.timePickerExpand;
-	}
-
 	closeTimePicker() {
-		this.timePickerExpand = false;
-		this.timePickerPresetsExpand = false;
+		this.popupExpanded.set('TimePicker', false);
+		this.popupExpanded.set('TimePickerPreset', false);
 	}
 
-	toggleTimePickerPresets() {
-		this.timePickerPresetsExpand = !this.timePickerPresetsExpand;
+	toggleExpander(popup: SearchPanelTitle) {
+		if (this.isDataInputsOk()) {
+			const newState = !this.popupExpanded.get(popup);
+			const isExpandIterator = this.popupExpanded.keys();
+			let isExpand = isExpandIterator.next();
+			while (!isExpand.done) {
```

Comment:
It seems simpler to use this.popupExpanded.foreach(...) instead of getting iterator + while

Reply:
you right